CONTRIBUTING.md

```diff
@@ -207,6 +207,96 @@ When updating dependencies, please only update the particular dependencies you
 need. Instead of running `go get -u=patch`, which will update anything, specify
 exactly the dependency you want to update.
 
+## Logging
+
+Operators, consensus engine and application developers all need information from
+the system while it is running. One avenue through which they get that
+information is via the logs. Whenever making contributions, please think
+carefully about what each of those groups of users would want to know about the
+operation of the system and try to adhere to the following guidelines as far as
+reasonably possible.
+
+### To log, or not to log
+
+Whether or not to log something at all should take into consideration how
+frequently the log message will appear. Users hate being spammed by large
+quantities of useless log messages. If you anticipate that a particular log
+message will occur frequently (e.g. a few times per minute), try to find ways to
+either eliminate that message or reduce its frequency (e.g. only logging every
+Nth message, or a summary message every minute or hour).
+
+### Log levels
+
+Different log levels should target different groups of users. At present, only
+**Debug**, **Info** and **Error** levels are supported.
+
+- **Debug**: Should primarily target consensus engine developers (i.e. core team
+  members and developers working on CometBFT forks).
+- **Info** and **Error**: Should primarily target operators and application
+  developers.
+
+### Log messages
+
+Log messages should always be tailored to the intended target audience. Unlike
+Go errors, **log messages must have the first letter of the message
+capitalized**. Only _errors_ in Go should start with a lowercase letter because
+they may end up being chained/embedded, but log messages are not chained in this
+same way.
+
+### Logging parameters
+
+In general, log messages should contain the **bare minimum** amount of
+information for those messages to be actionable by the target audience. So
+instead of dumping large quantities of raw data into the logs (e.g. RPC
+responses, transactions, block data), include lightweight references to the data
```

Comment:
Sounds valuable. Would this be similar to having a "trace" level? Or would "noise" messages be logged separately to the log messages? If they're logged separately, how does one link log messages to the logs to the noise messages?

Also, seems like a new type of feature we'd need to add, since our current logging framework doesn't support this.

Reply:
Yes, we don't support "noise" today (neither "warning", which seems like a miss to me TBH), so I take this convo as a discussion on possible improvements. My experience with "noise" in the past is that it was an super-verbose level (way more verbose than debug) were you dump all possible data to avoid having to attach a debugger (which is problematic with multithreaded or time sensitive processes).
You had to think twice before you enabled that level, just using it temporarily when debugging particular problems, and always restricted to one module only. Otherwise, boom!
Actually, a "noise" log level, combined with the idea I'm planning to work on as PI week (i described it in slack), would be super cool to have to say bye bye to dlv in most non-UT settings.